openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,198 @@
+"""ICCExternalEnrollment class file."""
+import json
+import logging
+from uuid import uuid4
+
+from django.conf import settings
+from rest_framework import status
+import requests
+import xmltodict
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.edxapp_wrapper.get_student import get_user
+from openedx_external_enrollments.models import EnrollmentRequestLog
+
+LOG = logging.getLogger(__name__)
+
+
+class ICCExternalEnrollment(BaseExternalEnrollment):
+    """
+    ICCExternalEnrollment class.
+    """
+    USER_CREATION_MAX_RETRIES = 5
+    USERNAME_SUFFIX_LENGTH = 5
+
+    def __str__(self):
+        return "ICC"
+
+    def _execute_post(self, url, data=None, headers=None, json_data=None):
+        """
+        Execute post request to achieve the ICC external enrollment.
+        """
+        response = requests.post(
+            url=url,
+            data=json_data,
+        )
+
+        return response
+
+    def _get_enrollment_headers(self):
+        """
+        Method that returns None by default, ICC integration does not headers.
+        """
+        return None
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Method that provide the relevant data for the enrollment.
+        """
+
+        enrollment_data = {}
+        try:
+            icc_user = self._get_icc_user(data)
+            enrollment_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_ENROLLMENT_API_FUNCTION,
+                "enrolments[0][roleid]": settings.ICC_LEARNER_ROLE_ID,
+                "enrolments[0][userid]": icc_user.get('id'),
+                "enrolments[0][courseid]": course_settings.get("external_course_run_id"),
+            }
+        except Exception:
+            pass
```

Comment:
Since this method returns the enrollment data on failure, it should not raise the exception? @ivanvgh

Reply:
The exception is handled in '_post_enrollment' from 'BaseExternalEnrollment'.

Reply:
I don't see where "_post_enrollment" handles this exception, I only see the call to this method anyway, I think that it's better to handle the exception where it is thrown because it will improve the debugging capabilities... @ivanvgh

Reply:
Alright, the exception was stored in the logs.